### PR TITLE
SNOW-1347387 Raise not implemented error for Series.dt.* methods

### PR DESCRIPTION
**Pull request by sfc-gh-nkumar**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1347387

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Raise not implemented error for Series.dt.* methods instead of falling back to native pandas.
   NOTE: I have kept the existing tests mostly unmodified even though there is redundancy. This can be useful when we implement these APIs.
   NOTE: This also require updates in docs/*supported.rst files. These files are not yet ported from snowpandas. So will do a follow up PR to update the docs.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 1]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
 
     @property
     def freq(self):
-        return self._query_compiler.dt_freq().to_pandas().squeeze()
+        return self._query_compiler.dt_property("freq").to_pandas().squeeze()
 
     def to_period(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_to_period(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("to_period"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1436]: This is a method not a property. I think it's better to add a corresponding method in `SnowflakeQueryCompiler` and raise a NotImplementedError there.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 2]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def to_pydatetime(self):
-        return Series(query_compiler=self._query_compiler.dt_to_pydatetime()).to_numpy()
+        return Series(
+            query_compiler=self._query_compiler.dt_property("to_pydatetime")
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1440]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 3]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
+        ).to_numpy()
 
     def tz_localize(self, *args, **kwargs):
-        return Series(
-            query_compiler=self._query_compiler.dt_tz_localize(*args, **kwargs)
-        )
+        return Series(query_compiler=self._query_compiler.dt_property("tz_localize"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1444]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 4]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def tz_convert(self, *args, **kwargs):
-        return Series(
-            query_compiler=self._query_compiler.dt_tz_convert(*args, **kwargs)
-        )
+        return Series(query_compiler=self._query_compiler.dt_property("tz_convert"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1447]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 5]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def normalize(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_normalize(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("normalize"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1450]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 6]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def strftime(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_strftime(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("strftime"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1453]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 7]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def round(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_round(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("round"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1456]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 8]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def floor(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_floor(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("floor"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1459]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 9]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def ceil(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_ceil(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("ceil"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1462]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 10]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def month_name(self, *args, **kwargs):
-        return Series(
-            query_compiler=self._query_compiler.dt_month_name(*args, **kwargs)
-        )
+        return Series(query_compiler=self._query_compiler.dt_property("month_name"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1465]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 11]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def day_name(self, *args, **kwargs):
-        return Series(query_compiler=self._query_compiler.dt_day_name(*args, **kwargs))
+        return Series(query_compiler=self._query_compiler.dt_property("day_name"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1468]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 12]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def total_seconds(self, *args, **kwargs):
-        return Series(
-            query_compiler=self._query_compiler.dt_total_seconds(*args, **kwargs)
-        )
+        return Series(query_compiler=self._query_compiler.dt_property("total_seconds"))
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1471]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 13]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
 
     def to_pytimedelta(self) -> "npt.NDArray[np.object_]":
-        res = self._query_compiler.dt_to_pytimedelta()
+        res = self._query_compiler.dt_property("to_pytimedelta")
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1474]: A method not a property.

File: src/snowflake/snowpark/modin/pandas/series_utils.py [hunk 14]
```diff
@@ -1428,90 +1430,82 @@ def tz(self) -> "tzinfo | None":
...
+        return Series(query_compiler=self._query_compiler.dt_property("start_time"))
 
     @property
     def end_time(self):
-        return Series(query_compiler=self._query_compiler.dt_end_time())
+        return Series(query_compiler=self._query_compiler.dt_property("end_time"))
 
     def to_timestamp(self, *args, **kwargs):
-        return Series(
-            query_compiler=self._query_compiler.dt_to_timestamp(*args, **kwargs)
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series_utils.py:1516]: `Series.dt.to_timestamp` doesn't exist in pandas docs. But I can see it in Modin OSS. Not sure if it's intentional or a mistake in Modin. @sfc-gh-mvashishtha any idea?

Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/pandas/series_utils.py:1516]: that looks like an oversight in Modin: https://github.com/modin-project/modin/issues/7232

File: src/snowflake/snowpark/modin/pandas/series_utils.py (hunk 1 above)
Comment (sfc-gh-nkumar) [reply src/snowflake/snowpark/modin/pandas/series_utils.py:1436]: Ah I missed that. Let me update this

**Review by sfc-gh-jkew — APPROVED** [review]

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Thanks for making all the changes!

**Pull request merged**